File: test-unit.php
use PHPUnit\Framework\TestCase;
use Mockery\MockInterface;
use Mockery;

class GenreControllerTest extends TestCase {

    /** @var IGenreService|MockInterface */
    private $genreService;

    /** @var GenreController */
    private $genreController;

    public function setUp(): void {
        $this->genreService = Mockery::mock(IGenreService::class);
        $this->genreController = new GenreController($this->genreService);
    }

    public function testGetAllGenres() {
        // Mocking
        $genres = [new Genre(), new Genre()];
        $this->genreService->shouldReceive('getAll')->andReturn($genres);

        // Test
        $response = $this->genreController->getAllGenres();

        // Verificación
        $this->assertEquals(200, $response->getStatusCode());
        $this->assertEquals($genres, $response->getBody());
    }

    // Otros métodos de prueba

    protected function tearDown(): void {
        Mockery::close();
    }
}

File: test-unit.py
import unittest
from unittest.mock import patch
from genre_controller import GenreController

class GenreControllerTest(unittest.TestCase):

    @patch('genre_controller.GenreService')
    def setUp(self, MockGenreService):
        self.genre_service = MockGenreService.return_value
        self.genre_controller = GenreController(self.genre_service)

    def test_get_all_genres(self):
        # Mocking
        genres = [Genre(), Genre()]
        self.genre_service.get_all.return_value = genres

        # Test
        response = self.genre_controller.get_all_genres()

        # Verificación
        self.assertEqual(response.status, 200)
        self.assertEqual(response.body, genres)

    # Otros métodos de prueba

if __name__ == '__main__':
    unittest.main()
